src/main.cpp

```diff
@@ -1655,6 +1656,100 @@ bool GetCertificate(const uint256 &hash, CScCertificate &certOut, uint256 &hashB
 }
 
 
+bool GetTxBaseObj(const uint256 &hash, std::unique_ptr<CTransactionBase>& pTxBase, uint256 &hashBlock, bool fAllowSlow)
```

Comment:
The implementation of `GetTxBaseObj` looks like you basically search if it is a tx and if you do not find it you try looking for certificate.
Can't you just make a call to `GetTransaction` and if it return false, try `GetCertificate`? This would avoid the code duplication

Reply:
If we stick to this implementation we might as well remove GetCertificate/GetTransaction, but if we choose to opt for the proposed approach we will leave it. As far as RPC cmds, we will probably remove getrawcertificate and decodedrawcertificate after a further check with client applications.